frame/curve-amm/src/tests.rs

```diff
@@ -229,3 +262,126 @@ fn exchange_test() {
+		assert_ne!(bob_lp_balance, 0);
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &BOB), 0);
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &BOB), 0);
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &CHARLIE), 0);
+		assert_ok!(CurveAmm::exchange(&CHARLIE, pool_id, 1, 0, alice_balance as u128, 0));
+		prop_assert_epsilon!(alice_balance as u128, Tokens::balance(MockCurrencyId::USDC, &CHARLIE));
+		Ok(())
+	}).unwrap();
+	}
+
+	#[test]
+	fn proptest_add_remove_liquidity(alice_balance in 0..u32::MAX,
+									 bob_balance in 0..u32::MAX) {
+
+	new_test_ext().execute_with(|| {
+		let assets = vec![MockCurrencyId::USDC, MockCurrencyId::USDT];
+		let amp_coeff = FixedU128::saturating_from_rational(1000i128, 1i128);
+		let fee = Permill::zero();
+		let admin_fee = Permill::zero();
+
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &ALICE), 0);
+		assert_ok!(Tokens::mint_into(MockCurrencyId::USDT, &ALICE, alice_balance.into()));
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &ALICE), alice_balance.into());
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &ALICE), 0);
+		assert_ok!(Tokens::mint_into(MockCurrencyId::USDC, &ALICE, alice_balance.into()));
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &ALICE), alice_balance.into());
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &BOB), 0);
+		assert_ok!(Tokens::mint_into(MockCurrencyId::USDT, &BOB, bob_balance.into()));
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &BOB), bob_balance.into());
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &BOB), 0);
+		assert_ok!(Tokens::mint_into(MockCurrencyId::USDC, &BOB, bob_balance.into()));
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &BOB), bob_balance.into());
+		let p = CurveAmm::create_pool(&ALICE, assets, amp_coeff, fee, admin_fee);
+		assert_ok!(&p);
+		let pool_id = p.unwrap();
+		let pool = CurveAmm::pool(pool_id);
+		assert!(pool.is_some());
+		let pool_lp_asset = CurveAmm::pool_lp_asset(pool_id);
+		assert!(pool_lp_asset.is_some());
+		let pool_lp_asset = pool_lp_asset.unwrap();
+		// 1 USDC = 1 USDT
+		let alice_amounts = vec![alice_balance as u128, alice_balance as u128];
+		assert_ok!(CurveAmm::add_liquidity(&ALICE, pool_id, alice_amounts.clone(), 0u128));
+		let alice_lp_balance = Tokens::balance(pool_lp_asset, &ALICE);
+		assert_ne!(alice_lp_balance, 0);
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &ALICE), 0);
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &ALICE), 0);
+		let pool = CurveAmm::pool(pool_id);
+		assert!(pool.is_some());
+		let bob_amounts = vec![bob_balance as u128, bob_balance as u128];
+		assert_ok!(CurveAmm::add_liquidity(&BOB, pool_id, bob_amounts.clone(), 0u128));
+		let bob_lp_balance = Tokens::balance(pool_lp_asset, &BOB);
+		assert_ne!(bob_balance, 0);
+		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &BOB), 0);
+		assert_eq!(Tokens::balance(MockCurrencyId::USDC, &BOB), 0);
+		let min_amt = vec![0u128, 0u128];
```

Comment:
Only nitpick I have here is that this is very difficult to follow, if you could create groups like this, and explain what you are checking/what stage of the procedure we're in

``
let a = 1;

// Test that arithmetic works
{
   assert_eq!(1, 1);
}

// Test that it works for zero
{ 
   assert_neq!(a, 0);
}
```